openquake/commonlib/logictree.py

```diff
@@ -1285,6 +1287,24 @@ def __init__(self, fname, tectonic_region_types=['*'], ltnode=None):
                 "'applyToTectonicRegionType' in %s" %
                 set(tectonic_region_types))
 
+    def check_imts(self, imts):
+        """
+        Make sure the IMTs are recognized by all GSIMs in the logic tree
+        """
+        for trt in self.values:
+            for gsim in self.values[trt]:
+                for attr in dir(gsim):
+                    coeffs = getattr(gsim, attr)
+                    if not isinstance(coeffs, CoeffsTable):
+                        continue
+                    for imt in imts:
+                        if imt.startswith('SA'):
+                            try:
+                                coeffs[from_string(imt)]
+                            except KeyError:
+                                raise ValueError('%s is not defined for %s' %
```

Comment:
It could be more useful to the user for the message to read "%s is outside of the period range supported by %s", as it tells the user explicitly why it was not possible to run the GMPE in this case